src/librustc/traits/query/normalize_erasing_regions.rs

```diff
@@ -56,7 +56,6 @@ impl<'tcx> TyCtxt<'tcx> {
     where
         T: TypeFoldable<'tcx>,
     {
-        assert!(!value.needs_subst());
```

Comment:
cc @eddyb @nikomatsakis 	I am not sure if we need this assertion. It may exist to catch bugs in other places in rustc, or it may just be overzealous invariant checking (I mean it was untriggerable until now).